### ISSUE TITLE
PHP fatal error (#2597)

### ISSUE THREAD

**Issue by tlarsh3** [woocommerce/woocommerce-paypal-payments#1977]
PHP Fatal error:  Uncaught Error: Class "WooCommerce\PayPalCommerce\Vaulting\VaultedCreditCardHandler" not found in /nas/content/live/website/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-vaulting/services.php:33#012Stack

**Comment by tlarsh3** [comment]
this was using 2.5.0 i had to rollback otherwise i would get non stop 502 gateway errors

**Comment by leandroprz** [comment]
I'm also getting a PHP fatal error after updating to v2.5.0:

```
Fatal error: Uncaught TypeError: WooCommerce\PayPalCommerce\WcSubscriptions\WcSubscriptionsModule::WooCommerce\PayPalCommerce\WcSubscriptions\{closure}(): Argument #1 ($methods) must be of type array, null given, called in D:\laragon\www\lp\wp-includes\class-wp-hook.php on line 324 and defined in D:\laragon\www\lp\wp-content\plugins\woocommerce-paypal-payments\modules\ppcp-wc-subscriptions\src\WcSubscriptionsModule.php:161 Stack trace: #0 D:\laragon\www\lp\wp-includes\class-wp-hook.php(324): WooCommerce\PayPalCommerce\WcSubscriptions\WcSubscriptionsModule->WooCommerce\PayPalCommerce\WcSubscriptions\{closure}(NULL) #1 D:\laragon\www\lp\wp-includes\plugin.php(205): WP_Hook->apply_filters(NULL, Array) #2 D:\laragon\www\lp\wp-content\plugins\woocommerce\includes\class-wc-payment-gateways.php(333): apply_filters('woocommerce_ava...', Array) #3 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\Tasks\Payments.php(93): WC_Payment_Gateways->get_available_payment_gateways() #4 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\Tasks\Payments.php(71): Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Payments::has_gateways() #5 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\Tasks\WooCommercePayments.php(125): Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Payments->is_complete() #6 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\TaskList.php(304): Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\WooCommercePayments->can_view() #7 [internal function]: Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskList->Automattic\WooCommerce\Admin\Features\OnboardingTasks\{closure}(Object(Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\WooCommercePayments)) #8 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\TaskList.php(305): array_filter(Array, Object(Closure)) #9 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\TaskList.php(171): Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskList->get_viewable_tasks() #10 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\TaskLists.php(349): Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskList->is_visible() #11 [internal function]: Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::Automattic\WooCommerce\Admin\Features\OnboardingTasks\{closure}(Object(Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskList)) #12 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Admin\Features\OnboardingTasks\TaskLists.php(350): array_filter(Array, Object(Closure)) #13 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Internal\Admin\Onboarding\OnboardingSetupWizard.php(217): Automattic\WooCommerce\Admin\Features\OnboardingTasks\TaskLists::get_visible() #14 D:\laragon\www\lp\wp-includes\class-wp-hook.php(324): Automattic\WooCommerce\Internal\Admin\Onboarding\OnboardingSetupWizard->component_settings(Array) #15 D:\laragon\www\lp\wp-includes\plugin.php(205): WP_Hook->apply_filters(Array, Array) #16 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Internal\Admin\WCAdminSharedSettings.php(61): apply_filters('woocommerce_adm...', Array) #17 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Blocks\Assets\AssetDataRegistry.php(268): Automattic\WooCommerce\Internal\Admin\WCAdminSharedSettings->Automattic\WooCommerce\Internal\Admin\{closure}() #18 D:\laragon\www\lp\wp-content\plugins\woocommerce\src\Blocks\Assets\AssetDataRegistry.php(393): Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry->execute_lazy_data() #19 D:\laragon\www\lp\wp-includes\class-wp-hook.php(324): Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry->enqueue_asset_data('') #20 D:\laragon\www\lp\wp-includes\class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array) #21 D:\laragon\www\lp\wp-includes\plugin.php(517): WP_Hook->do_action(Array) #22 D:\laragon\www\lp\wp-admin\admin-footer.php(95): do_action('admin_print_foo...') #23 D:\laragon\www\lp\wp-admin\index.php(212): require_once('D:\\laragon\\www\\...') #24 {main} thrown in D:\laragon\www\lp\wp-content\plugins\woocommerce-paypal-payments\modules\ppcp-wc-subscriptions\src\WcSubscriptionsModule.php on line 161

Ha habido un error crítico en esta web. Por favor, comprueba la bandeja de entrada del correo electrónico del administrador de tu sitio para obtener instrucciones.
```

I'm only seeing that at the bottom of my WP admin pages, not on the frontend.

I disabled the Woo Subscriptions plugin, but the error still shows up.

Edit: the error message disappears when disabling the plugin [Mobbex Subscriptions for WooCommerce](https://github.com/mobbexco/woocommerce-subscriptions).

**Comment by Sirvijver** [comment]
Hello @leandroprz and @tlarsh3,

we can confirm this issue and have opened a bug for our developers to look into this.
For another issue, we have re-released version 2.4.3 as release 2.5.1, so that issue would not persist for the merchants using our plugin. This action also temporarily resolved this issue, so if you update to 2.5.1, you will have the same situation as you have now, using 2.4.3.

I have linked this thread to our developers bug, so it should get updated when a fix is available, for now, using version 2.5.1 should not cause any issues for you. Let me know if you have any questions.

Kind regards, 
Joost

**Comment by tlarsh3** [comment]
@Sirvijver that looks like it resolved one issue however my error logger is saying it has created another problem after i upgraded to 2.5.1:

Fatal error: Trait &quot;WooCommerce\PayPalCommerce\Subscription\FreeTrialHandlerTrait&quot; not found in /nas/content/live/website/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-button/src/Endpoint/CreateOrderEndpoint.php on line 43

**Comment by Sirvijver** [comment]
Hello @tlarsh3 

The 2.5.1 release was a revert back to 2.4.3 in order to resolve this issue. Next Monday we expect to release version 2.5.2, that should fix the problem that was caused with 2.5.0. Later today I expect a beta version to be available here, so if you want, you can check if that resolved this issue as well. If you decide to test, we would appreciate your feedback (good or bad).
Thank you for your patience.

Kind regards, 
Joost

**Comment by InpsydeNiklas** [comment]
This issue was addressed in #1985 & included in the latest [2.5.2 update](https://github.com/woocommerce/woocommerce-paypal-payments/releases).
